django/utils/deprecation.py

```diff
@@ -140,3 +144,25 @@ async def __acall__(self, request):
                 thread_sensitive=True,
             )(request, response)
         return response
+
+
+def adjust_stacklevel_for_warning(skip_file_prefixes):
+    def _get_non_django_stacklevel():
+        django_path = os.path.dirname(django.__file__)
+        stacklevel = 1
+        # Exclude current and nested function frames with [2:].
+        for frame_info in inspect.stack()[2:]:
+            filename = os.path.abspath(frame_info.filename)
+            if not filename.startswith(django_path):
+                return stacklevel
+            stacklevel += 1
+        return 1
```

Comment:
nit: I don't think this needs to be definied inside of `adjust_stacklevel_for_warming`? We could just define this as a top-level (underscored) method that gets called right? Since we don't seem to be reproducing the `skip_file_prefixes` logic anyways